ui/pages/settings/networks-tab/networks-form/networks-form.js

```diff
@@ -618,30 +619,29 @@ const NetworksForm = ({
             networkConfigurationId,
           }),
         );
-
-        trackEvent({
-          event: MetaMetricsEventName.CustomNetworkAdded,
-          category: MetaMetricsEventCategory.Network,
-          properties: {
-            block_explorer_url: blockExplorerUrl,
-            chain_id: prefixedChainId,
-            network_name: networkName,
-            source_connection_method:
-              MetaMetricsNetworkEventSource.CustomNetworkForm,
-            token_symbol: ticker,
-          },
-        });
-
-        submitCallback?.();
       }
+      trackEvent({
```

Comment:
Hmm, any idea why this was previously only called if `addNewNetwork` was true? I'm having trouble understanding what `addNewNetwork` is supposed to signify. Really wish the props were commented for this component.

Reply:
Yeah its pretty confusing... It seems to me that its used at the root to [differentiate between the `NETWORKS_ROUTE` and `ADD_NETWORK_ROUTE`](https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/settings/settings.component.js#L373-L382) which both render the `NetworksTab` component. The `addNewNetwork` is then used for [this ternary](https://github.com/MetaMask/metamask-extension/blob/65f5b81093300e0dba1a5344d28fa19528b739ed/ui/pages/settings/networks-tab/networks-tab.js#L123-L132) where `NetworksForm` is rendered  if `addNewNetwork` is truthy and otherwise `NetworksTabContent` is rendered if its falsy. Basically in UX terms this seems to enable the following path:

https://github.com/MetaMask/metamask-extension/assets/34557516/ff7a1bf5-a847-47db-9acc-46aadfb77ac6

where the network form is rendered but out of the context of the full network settings seen here:
![Screenshot 2024-02-07 at 9 56 16 AM](https://github.com/MetaMask/metamask-extension/assets/34557516/bc562832-5a25-48f8-9491-e548b30fe7d7)

in this context a network can only be added not edited or deleted. I guess that's why its called `addNewNetwork`. Obviously this flow could use some cleanup. But for now I would say this trackEvent almost certainly belongs [in this `else` block](https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/settings/networks-tab/networks-form/networks-form.js#L593-L613) since that is where a custom network is added.

Reply:
```
if (addNewNetwork) {
        dispatch(
          setNewNetworkAdded({
            nickname: networkName,
            networkConfigurationId,
          }),
        );
      }
```

can probably live in that else block as well since it appears to be mutually exclusive with the `selectedNetwork` prop. But I'm going to leave that alone for now. Not looking to let this creep too much.

Reply:
Yeah, it looks to me like it belongs in that else block as well.

Understood if you'd rather not fix that in this PR, but as it stands this PR seems to break it in a different way; we're now sending this event even for edits (if I understand this correctly).

Reply:
No I'm happy to move it into the else block. Will do that now.

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/22832/commits/3c770ffb068df8590567c2cbece252c0f2d9f1b8